### DISCUSSION TITLE
Pasive and Active spells

### DISCUSSION THREAD

**Discussion by Kamusvalenzuela** [AscensionGameDev/Intersect-Engine#1499] [Ideas]
**Is your feature request related to a problem? Please describe.**
The possibility to activate some spells consuming the mana per second, or pasive spells for some classes that will give more ap or ad for example. i think that will be a really nice feature for all mmorpgs proyects

**Describe the solution you'd like**
-Posibility to do pasive spells
-Posibility to do active spells who will consume the maná or life per second

**Comment by WeylonSantana** [comment]
This one has a lot of likes, which should be converted into votes
But how would it work? think about the intersection itself? how should it be implemented?

**Reply by Kamusvalenzuela** [reply]
![Screenshot_446](https://user-images.githubusercontent.com/51339474/192190796-b7ee4f05-2810-4a51-b55d-56b6248710a2.jpg)
The pasive spells can modify only the players stats and damage (Like a combat spell but without the HP/MP costs and without the possibility of "disable it", i.e: Pasive spells for "X" class)
![Screenshot_447](https://user-images.githubusercontent.com/51339474/192192445-1e4611bb-144d-4192-bbda-69fc3f961b02.jpg)
The activable spells can be a pasive spell modificated (With a few changes) 
If the checkbox "Activable" is enable -> we will enable the form "HP/MP costs" without the cooldowns
If the checkbox "Activable" is dissabled -> We only have the modifiers and the hp/mp damage only (For make some buffs with pros and negatives values)

**Reply by WeylonSantana** [reply]
And about duration? Infinity? About multiple cast,  cumulative? And how track how many passive you currently have?

**Reply by Arufonsu** [reply]
cast? the whole point of passive skills is to have an ability that provides some static stat modifier/buff.
some of them could be triggered after different actions/conditions, while some of them could be simply added permanently as static stat buffs. we could somehow calculate the additional stat buffs upon login depending on the total value that's provided by the skills themselves.
Also, graphically around the char window we might want to denote the base stats from the additional passive spells/skills buffs along with the temporary buffs provided by consumables. (10 + 22) (base stat (10) + (passive skill provided buff (11) + consumable temporary buff (11)) ) or something like that, if that's not already a thing yet.


About the "activable" thing though, i sort of disagree with that, it would turn the passive skill/spell into an active one, which we already have within intersect.

**Comment by Arufonsu** [comment]
something to consider:

![image](https://user-images.githubusercontent.com/17498701/192364375-8a659be1-9266-4ed1-852b-f912490e1bdb.png)
